pkg/build/webhook/generic/generic.go

```diff
@@ -66,6 +68,9 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 			glog.V(4).Infof("Error unmarshaling json %v, but continuing", err)
 			return nil, true, nil
 		}
+		if len(data.Env) > 0 {
+			updateBuildEnv(buildCfg, data.Env)
+		}
```

Comment:
I think there are a couple problems w/ this approach:
1) you're modifying the buildconfig whereas ideally we'd modify the build object
2) you're modifying the buildconfig as a side-effect (it's not obvious from the function definition that the buildconfig which gets passed in is going to get mutated)

I think you might want to take a look at the approach I used here:
https://github.com/openshift/origin/pull/8405/files